packages/flutter_tools/lib/src/context_runner.dart

```diff
@@ -88,11 +89,12 @@ Future<T> runInContext<T>(
     body: runnerWrapper,
     overrides: overrides,
     fallbacks: <Type, Generator>{
-      Analytics: () => Analytics(
-        tool: DashTool.flutterTool,
-        flutterChannel: globals.flutterVersion.channel,
-        flutterVersion: globals.flutterVersion.frameworkVersion,
-        dartVersion: globals.flutterVersion.dartSdkVersion,
+      Analytics: () => getAnalytics(
+        runningOnBot: runningOnBot,
+        flutterVersion: globals.flutterVersion,
+        suppressEnvFlag:
+            globals.platform.environment['FLUTTER_SUPPRESS_ANALYTICS'] ==
+                'true',
       ),
```

Comment:
it occurs to me, this is already a function, you can just inline the logic here:

```suggestion
      Analytics: () {
        final String version = flutterVersion.getVersionString(redactUnknownBranches: true);
	if (
          // Ignore local user branches.
          version.startsWith('[user-branch]') ||
          // Many CI systems don't do a full git checkout.
          version.endsWith('/unknown') ||
          // Ignore bots.
          runningOnBot ||
          // Ignore when suppressed by FLUTTER_SUPPRESS_ANALYTICS.
          suppressEnvFlag,
        ) {
          return NoOpAnalytics();
        }
        return Analytics(
          tool: DashTool.flutterTool,
          flutterChannel: flutterVersion.channel,
          flutterVersion: flutterVersion.frameworkVersion,
          dartVersion: flutterVersion.dartSdkVersion,
        );
      },
```

Reply:
also note you must follow the flutter style guide here

Reply:
Lol actually if you look at the first commit on this PR, I had it like your suggestion but I used the function instead to match the other generators a bit more. But good to know I can have the logic in the file